tests/posix/fs/src/test_fs_file.c

```diff
@@ -287,6 +300,17 @@ ZTEST(posix_fs_file_test, test_fs_unlink)
 	zassert_true(test_file_delete() == TC_PASS);
 }
 
+/**
+ * @brief Test for POSIX remove API
+ *
+ * @details Test deletes a file through POSIX remove API.
+ */
+ZTEST(posix_fs_file_test, test_fs_remove)
+{
+	zassert_true(test_file_open() == TC_PASS);
+	zassert_true(test_file_remove() == TC_PASS);
+}
+
```

Comment:
/* FIXME: restructure tests as per #46897 */ ; can we start here by not doing that and directly use
`zassert_equal()`, `zassert_true()` here wit assert message?
Also, shouldn't negative tests be also here, like incorrect path provided or non existing mount point path?

Reply:
Yeah, I agree - let's let the dinosaurs die (i.e. start writing good tests and have less to refactor later).

Definitely want to validate behaviour in the case of negative / degenerate input as well.